pkg/cli/admin/mustgather/mustgather.go

```diff
@@ -649,6 +649,7 @@ func (o *MustGatherOptions) newPod(node, image string) *corev1.Pod {
 			TerminationGracePeriodSeconds: &zero,
 			Tolerations: []corev1.Toleration{
 				{
+					Key:      "node-role.kubernetes.io/master",
 					Operator: "Exists",
```

Comment:
This particular use-case is to ensure that we match all possible taints, see Note section in https://kubernetes.io/docs/concepts/scheduling-eviction/taint-and-toleration/ I'd suggest adding another toleration like the one you're proposing, but only when node wasn't specified. Your current code will break that last assumption.

Reply:
Thank you @soltysh. I have modified the toleration considering the conditions. Let me know if this fits better or not.